### PR TITLE
Change language picker icon

### LINKED ISSUE #3372: Change language-picker icon so it's not the Google Translate icon
I saw that a language picker was added in #2294, that's great!

However, I noticed that the language picker icon uses the Google Translate icon:
Google Translate icon

I think this makes it look like the site will be automatically translated with Google Translate, rather than actual human translations.

### PR DESCRIPTION
**Pull request by ArkinSolomon**
Fixes #3372. I know this had 2 pull requests created already, with the most recent one dying pretty quickly. It was a pretty simple change to use the new image. I did slightly modify the icon to be white only. It works fine for both light and dark modes.

Screen Shot 2022-03-18 at 1 50 32 PM

icon128px-exported-black

### CODE REVIEW

**Comment by Trott** [comment]
Where did the icon come from originally?

**Comment by ArkinSolomon** [comment]
It came from http://www.languageicon.org/, see #3503.

**Comment by Trott** [comment]
@nodejs/website

**Comment by XhmikosR** [comment]
It shouldn't be a PNG.

**Comment by XhmikosR** [comment]
It shouldn't be a PNG.

**Review by XhmikosR — CHANGES_REQUESTED** [review]
Please use SVG files after optimizing it and no redundant attributes.

**Comment by ArkinSolomon** [comment]
I have changed the icons to an svg by converting it from `.eps`, and manually added light and dark variants and used css to change them when the theme changes.

File: layouts/css/page-modules/_header.scss [hunk 1]
```diff
@@ -3,7 +3,7 @@ header {
   text-align: center;
   position: relative;
 
-  > .container {
+  >.container {
```
Comment (XhmikosR) [inline layouts/css/page-modules/_header.scss:6]: unrelated change

File: layouts/partials/header.hbs [hunk 2]
```diff
@@ -26,10 +26,8 @@
...
-        </svg>
+        <img class="lang-picker-icon light-icon" src="/static/images/lang-picker-icon-light.svg" alt="Language">
```
Comment (XhmikosR) [inline layouts/partials/header.hbs]: You shouldn't use `img` tags. Instead inline them like before. Then add a class and change the `fill` value only conditionally. That way you can use one icon.

File: static/images/lang-picker-icon-dark.svg [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (XhmikosR) [inline static/images/lang-picker-icon-dark.svg]: these are unoptimized

File: static/images/lang-picker-icon-dark.svg [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+<svg xmlns:dc="http://purl.org/dc/elements/1.1/"
+  xmlns:cc="http://creativecommons.org/ns#"
+  xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
+  xmlns:svg="http://www.w3.org/2000/svg"
+  xmlns="http://www.w3.org/2000/svg"
+  xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
+  xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape" version="1.1" id="svg2" xml:space="preserve" width="3214.9468" height="3725.3333" viewBox="0 0 3214.9468 3725.3333" sodipodi:docname="Language-Icon-Fireworks [Converted].eps">
```
Comment (XhmikosR) [inline static/images/lang-picker-icon-dark.svg]: Also, the icons could probably be smaller

File: layouts/partials/header.hbs (hunk 2 above)
Comment (XhmikosR) [reply layouts/partials/header.hbs:29]: Then add a class and change the `fill` value only conditionally. That way you can use one icon.

File: layouts/css/layout/_dark-theme.scss [hunk 5]
```diff
@@ -146,4 +146,11 @@
       background-color: $dark-code-background;
     }
   }
+
+  .light-icon {
+    display: none;
```
Comment (XhmikosR) [inline layouts/css/layout/_dark-theme.scss]: This and display: inline should be moved to [utils](https://github.com/nodejs/nodejs.org/blob/main/layouts/css/_utils.scss) like:

```css
.d-inline {
  display: inline !important;
}

.d-none {
  display: none !important;
}
```

That way you can reuse them across the codebase.

File: layouts/partials/header.hbs (hunk 2 above)
Comment (XhmikosR) [reply layouts/partials/header.hbs]: Example optimized SVG which you can then add a class and toggle the fill rule accordingly instead of having two icons:

```xml
<svg xmlns="http://www.w3.org/2000/svg" xml:space="preserve" width="3214.947" height="3725.333" viewBox="0 0 3214.947 3725.333">
  <path fill="none" stroke="#fff" stroke-width="53.332" d="M1606.093 478.88 362.44 39.947v2706.786l1243.653-402.36V478.88"/>
  <path fill="#333" d="M1576.24 477.52 2867.507 38.6v2706.787l-1291.267-402.36V477.52M39.947 3220.573l1536.293-512.066V477.227L39.947 989.293v2231.28"/>
  <path fill="#fff" d="m2276.28 3254.28 217.733 358.213 114.814-332.76ZM562.32 1246.333c-8.16-8.013 10.627 65.48 36.773 91.92 46.36 46.774 82.574 52.8 101.854 53.574 42.666 1.706 95.32-10.627 126.586-23.734 30.254-12.906 83.267-39.973 103.334-79.453m406.6 836.253c-12.107-4.4-262.587-108.16-298.094-125.16-29.053-13.973-100.293-44.093-133.813-57.773 94.413-145.573 154.013-255.427 161.947-272.16 14.68-30.613 114.613-226.16 116.946-238.2 2.267-12.2 5.107-57.267 2.907-67.973-2.2-10.92-38.84 10.066-88.587 26.933-49.826 16.8-144.52 78.387-181.093 86.107-36.707 7.653-154.013 52.093-214.04 72.013-60.027 19.92-173.573 54.573-220.28 67.187-46.773 12.613-87.6 13.613-113.76 21.546 0 0 3.48 36.64 10.427 47.627 6.866 10.987 31.613 37.92 60.386 45.44 28.774 7.573 76.4 4.533 98.094-.427 21.68-5.04 59.24-23.386 64.28-31.4 5.093-8.146-2.627-33.24 5.946-40.826 8.654-7.507 122.974-34.227 166.134-47.267 43.16-13.267 208.373-70.173 230.773-67.267-7.093 23.534-139.987 286.76-182.787 365.294-42.813 78.52-291.506 423.973-344.453 484.853-40.187 46.28-137.573 164.707-171.307 191.427 8.507 2.346 68.814-2.827 79.8-9.627 68.467-42.173 182.507-184.133 219.227-227.373 109.147-128 205.04-262.454 281.08-377.84h.08c14.813 6.173 134.587 103.76 165.84 125.386 31.253 21.614 154.587 90.427 181.307 101.84 26.72 11.56 129.413 58.907 133.733 42.88 4.32-16.16-18.573-110.64-30.693-115.24"/>
  <path fill="#fff" fill-rule="evenodd" d="M758.667 3429.333c24 14.667 46.666 26.667 72 38.667 50.666 25.333 108 52 162.666 72 74.667 28 149.334 50.667 224 68 41.334 9.333 86.667 17.333 130.667 24 4 0 122.667 14.667 146.667 14.667h120c46.666-4 90.666-6.667 137.333-13.334 37.333-5.333 78.667-12 118.667-21.333 29.333-6.667 60-13.333 89.333-22.667 28-8 60-18.666 90.667-29.333 20-6.667 41.333-16 62.666-24 17.334-8 38.667-17.333 58.667-25.333 24-10.667 52-25.334 78.667-38.667 21.333-10.667 45.333-24 68-37.333 17.333-9.334 57.333-40 78.666-40 24 0 40 21.333 40 40 0 38.666-52 50.666-76 68-25.333 17.333-56 30.666-82.666 45.333-53.334 28-108 52-160 72-68 25.333-142.667 49.333-209.334 65.333-25.333 5.334-50.666 12-76 16-13.333 2.667-152 24-190.666 24h-176c-46.667-4-96-9.333-142.667-16-41.333-6.666-85.333-14.666-126.667-24-32-6.666-66.666-16-97.333-25.333-53.333-14.667-105.333-33.333-156-53.333-92-34.667-188-80-278.667-140-16-10.667-17.333-21.334-17.333-33.334 0-20 14.667-38.666 38.667-38.666 21.333 0 64 30.666 72 34.666M1616 472v2240c-1.333 6.667-4 13.333-9.333 20-2.667 4-8 9.333-12 10.667C1561.333 2756 60 3260 40 3260c-16 0-30.667-10.667-38.667-28 0-1.333-1.333-2.667-1.333-5.333V985.333c2.667-6.666 4-16 9.333-21.333C20 949.333 38.667 946.667 50.667 942.667c22.666-8 1501.333-504 1522.666-504C1586.667 438.667 1616 448 1616 472zm-81.333 2206.667-1453.334 484v-2144l1453.334-484v2144"/>
  <path fill="#fff" fill-rule="evenodd" d="M2908 42.667V2740c-1.333 30.667-22.667 44-42.667 44-17.333 0-142.666-42.667-164-49.333-168-52-337.333-104-504-156-37.333-12-76-24-112-36-32-9.334-66.666-20-98.666-30.667-142.667-44-288-88-430.667-136-5.333-1.333-18.667-20-18.667-24V468c2.667-6.667 5.334-14.667 12-20 10.667-12 468-164 648-224 48-17.333 649.334-224 668-224 24 0 42.667 17.333 42.667 42.667zm-81.333 2646.666-1209.334-376V506.667L2826.667 96v2593.333"/>
  <path fill="#fff" d="m3214.947 3254.28-1615.614-514.947 6.76-2241.733 1608.854 512.067V3254.28"/>
  <path fill="#333" fill-rule="evenodd" d="m2305.307 1137.587 208.146 63.053 379.214 1366.693-213.8-64.866-76.814-280.574-441.84-133.92-95.013 228.56-213.867-64.88zm95.16 361.813-158.574 383.28 291.534 88.373z"/>
</svg>
```

I kept the line breaks but they can also be removed later if we want to.

Comment (ArkinSolomon) [reply layouts/partials/header.hbs]: I ran it through https://jakearchibald.github.io/svgomg/, which optimized it I think. Then put it directly into the HTML. I used CSS to dynamically change the fill for two of the elements for dark theme.

File: static/images/lang-picker-icon-dark.svg (hunk 4 above)
Comment (ArkinSolomon) [reply static/images/lang-picker-icon-dark.svg]: Do you mean smaller as in decreasing the view box? I have fixed it to have a height of 25px, but the view box is the same.

File: layouts/css/layout/_dark-theme.scss (hunk 5 above)
Comment (ArkinSolomon) [reply layouts/css/layout/_dark-theme.scss]: This CSS is no longer used, only using one class to change the fill color on some portions of the SVG.

File: layouts/partials/header.hbs [hunk 6]
```diff
@@ -26,9 +26,14 @@
...
+        <svg xmlns="http://www.w3.org/2000/svg" xml:space="preserve" height="25" viewBox="0 0 3214.947 3725.333">
```
Comment (XhmikosR) [inline layouts/partials/header.hbs]:
```xml
<svg xmlns="http://www.w3.org/2000/svg" xml:space="preserve" height="25" viewBox="0 0 3214.947 3725.333">
  <path fill="none" stroke="#fff" stroke-width="66.665" d="M1606.053 478.961 362.431 40.04v2706.719l1243.622-402.35V478.96"/>
  <path d="M1576.2 477.601 2867.436 38.692v2706.72l-1291.234-402.35V477.601M39.946 3220.585 1576.2 2708.532V477.308L39.946 989.362v2231.224" class="header-background-fill"/>
  <path fill="#fff" d="m2276.223 3254.292 217.728 358.204 114.81-332.752ZM562.306 1246.395c-8.16-8.013 10.626 65.479 36.772 91.918 46.36 46.772 82.572 52.799 101.851 53.572 42.666 1.707 95.318-10.626 126.584-23.733 30.252-12.906 83.264-39.972 103.33-79.451m406.59 836.232c-12.106-4.4-262.58-108.157-298.086-125.157-29.052-13.973-100.29-44.092-133.81-57.771 94.411-145.57 154.01-255.42 161.943-272.154 14.68-30.612 114.61-226.154 116.944-238.194 2.266-12.2 5.106-57.265 2.906-67.971-2.2-10.92-38.839 10.066-88.584 26.932-49.826 16.8-144.517 78.385-181.089 86.105-36.706 7.653-154.01 52.092-214.035 72.011-60.025 19.92-173.569 54.572-220.274 67.185-46.772 12.613-87.598 13.613-113.757 21.546 0 0 3.48 36.64 10.426 47.626 6.867 10.986 31.613 37.919 60.385 45.439 28.773 7.573 76.398 4.533 98.091-.427 21.68-5.04 59.239-23.386 64.279-31.399 5.093-8.146-2.627-33.24 5.946-40.826 8.653-7.506 122.97-34.225 166.13-47.265 43.158-13.266 208.367-70.172 230.767-67.265-7.093 23.533-139.983 286.753-182.782 365.284-42.813 78.518-291.5 423.963-344.445 484.841-40.186 46.28-137.57 164.703-171.302 191.422 8.506 2.347 68.811-2.826 79.798-9.626 68.465-42.173 182.502-184.129 219.22-227.368 109.145-127.997 205.036-262.447 281.074-377.83h.08c14.813 6.173 134.583 103.757 165.836 125.383 31.252 21.613 154.582 90.424 181.302 101.838 26.72 11.56 129.41 58.905 133.73 42.878 4.32-16.16-18.573-110.637-30.693-115.237"/>
  <path fill="#fff" fill-rule="evenodd" d="M758.648 3429.34c24 14.667 46.665 26.667 71.998 38.666 50.665 25.333 107.997 52 162.663 71.999 74.664 27.999 149.33 50.665 223.994 67.998 41.332 9.333 86.664 17.333 130.663 24 4 0 122.664 14.666 146.663 14.666h119.997c46.666-4 90.665-6.667 137.33-13.333 37.333-5.334 78.665-12 118.664-21.333 29.333-6.667 59.998-13.333 89.331-22.666 28-8 59.998-18.667 90.664-29.333 20-6.666 41.333-16 62.666-24 17.332-8 38.665-17.332 58.665-25.332 24-10.666 51.998-25.333 78.664-38.666 21.333-10.666 45.333-24 67.999-37.332 17.333-9.333 57.332-40 78.664-40 24 0 40 21.334 40 40 0 38.666-52 50.665-75.999 67.998-25.332 17.333-55.998 30.666-82.664 45.332-53.332 28-107.998 52-159.996 71.999-67.999 25.332-142.663 49.332-209.328 65.331-25.333 5.334-50.666 12-75.999 16-13.333 2.667-151.996 24-190.661 24H1466.63c-46.666-4-95.998-9.334-142.663-16-41.332-6.667-85.331-14.667-126.664-24-31.999-6.666-66.665-16-97.33-25.332-53.333-14.667-105.331-33.333-155.997-53.332-91.997-34.666-187.995-79.998-278.66-139.997-15.999-10.666-17.332-21.333-17.332-33.332 0-20 14.666-38.666 38.665-38.666 21.333 0 63.999 30.666 71.999 34.666M1615.96 472.08v2239.944c-1.334 6.667-4 13.333-9.333 20-2.667 4-8 9.333-12 10.666-33.333 13.333-1534.629 517.32-1554.628 517.32-16 0-30.666-10.666-38.666-27.999 0-1.333-1.333-2.666-1.333-5.333V985.402c2.667-6.667 4-16 9.333-21.333C20 949.403 38.666 946.736 50.665 942.736c22.667-8 1501.296-503.987 1522.629-503.987 13.333 0 42.666 9.333 42.666 33.332zm-81.332 2206.612L81.331 3162.68V1018.734l1453.297-483.988v2143.947"/>
  <path fill="#fff" fill-rule="evenodd" d="M2907.927 42.759v2697.266c-1.333 30.665-22.666 43.998-42.665 43.998-17.333 0-142.663-42.665-163.996-49.332-167.996-51.998-337.325-103.997-503.988-155.996-37.332-12-75.998-24-111.997-35.999-31.999-9.333-66.665-20-98.664-30.666-142.663-43.999-287.993-87.998-430.656-135.996-5.333-1.334-18.666-20-18.666-24V468.081c2.667-6.666 5.333-14.666 12-20 10.666-11.999 467.988-163.995 647.983-223.994 48-17.332 649.318-223.994 667.984-223.994 24 0 42.665 17.333 42.665 42.666zm-81.331 2646.6-1209.303-375.99V506.747L2826.596 96.091v2593.268"/>
  <path fill="#fff" d="m3214.866 3254.292-1615.573-514.934 6.76-2241.677 1608.813 512.054v2244.557"/>
  <path fill-rule="evenodd" d="m2305.249 1137.651 208.141 63.052 379.204 1366.66-213.794-64.866-76.812-280.566-441.829-133.917-95.01 228.555-213.862-64.879zm95.158 361.805-158.57 383.27 291.526 88.371z" class="header-background-fill"/>
</svg>
```

This is the optimized SVG I get. Please confirm it's fine and if so use that. Although, `width` is missing

Comment (ArkinSolomon) [reply layouts/partials/header.hbs]: Done, using this SVG in the new commit

Comment (XhmikosR) [reply layouts/partials/header.hbs]: I'd still add width and make the icon smaller since there's no need to have such a huge viewbox

File: layouts/partials/header.hbs [hunk 7]
```diff
@@ -26,9 +26,15 @@
...
+          <path d="M21810 27620V7390c-10-230-170-330-320-330-130 0-1070 320-1230 370-1260 390-2530 780-3780 1170-280 90-570 180-840 270-240 70-500 150-740 230-1070 330-2160 660-3230 1020-40 10-140 150-140 180v14130c20 50 40 110 90 150 80 90 3510 1230 4860 1680 360 130 4870 1680 5010 1680 180 0 320-130 320-320zm-610-19850-9070 2820v13550l9070 3080V7770" style="fill:#fff;fill-opacity:1;fill-rule:evenodd;stroke:none" transform="matrix(.13333 0 0 -.13333 0 3725.333)"/>
+          <path d="M24112.1 3532.9 11995 7395l50.7 16813 12066.4-3840.5V3532.9" style="fill:#fff;fill-opacity:1;fill-rule:nonzero;stroke:none" transform="matrix(.13333 0 0 -.13333 0 3725.333)"/>
+          <path d="m17289.8 19408.1 1561.1-472.9L21695 8685l-1603.5 486.5-576.1 2104.3-3313.8 1004.4L15489 10566l-1604 486.6zm713.7-2713.6-1189.3-2874.6 2186.5-662.8z" class="header-background-fill" style="fill-opacity:1;fill-rule:evenodd;stroke:none" transform="matrix(.13333 0 0 -.13333 0 3725.333)"/>
+        </svg>
```
Comment (osk2) [inline layouts/partials/header.hbs]: Please remove redundant `</svg>`.

**Review by osk2 — CHANGES_REQUESTED** [review]

File: layouts/partials/header.hbs (hunk 7 above)
Comment (ArkinSolomon) [reply layouts/partials/header.hbs]: Didn't notice that, removed.

File: layouts/partials/header.hbs (hunk 6 above)
Comment (ArkinSolomon) [reply layouts/partials/header.hbs]: I have shrunk the view box width to 512, and scaled the view box heigh accordingly, as well as scaled everything else down by adding a `<g>` tag with the scale transform and running it through the same SVG optimizer.

**Review by osk2 — APPROVED** [review]
LGTM

**Comment by ArkinSolomon** [comment]
Is this PR good for merging @XhmikosR ?

**Review by aymen94 — APPROVED** [review]

**Comment by XhmikosR** [comment]
Sorry, I can't review the result (this is one of the cases PR previews would go a long way).

If someone can confirm everything looks good on both themes, that's good enough from my side judging by the code changes.

**Comment by Trott** [comment]
> Sorry, I can't review the result (this is one of the cases PR previews would go a long way).

@XhmikosR If you're online right now, here's a preview I'll keep around for the next 30 minutes or so: https://8080-arkinsolomon-nodejsorg-yges5368tg1.ws-us38.gitpod.io/

**Comment by XhmikosR** [comment]
Thanks @Trott!

Confirmed LGTM. The same optimizations could be made to the dark switcher icons later (inline them, optimize them, toggle the `d-none` class etc)

**Review by XhmikosR — APPROVED** [review]
